src/main/java/org/terasology/lost/OnSpawnSystem.java

```diff
@@ -15,26 +15,68 @@
  */
 package org.terasology.lost;
 
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityBuilder;
 import org.terasology.entitySystem.entity.EntityManager;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.logic.inventory.InventoryComponent;
 import org.terasology.logic.inventory.InventoryManager;
 import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.math.Side;
 import org.terasology.registry.In;
+import org.terasology.logic.console.Console;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.world.WorldProvider;
+import org.terasology.structureTemplates.events.SpawnStructureEvent;
+import org.terasology.structureTemplates.util.BlockRegionTransform;
+
+import static org.terasology.lost.LevelSpawnSystem.getGroundHeight;
 
 @RegisterSystem(RegisterMode.AUTHORITY)
 public class OnSpawnSystem extends BaseComponentSystem {
     @In
     private EntityManager entityManager;
     @In
     private InventoryManager inventoryManager;
+    @In
+    private Console console;
+    @In
+    private AssetManager assetManager;
+    @In
+    private WorldProvider worldProvider;
 
     @ReceiveEvent
     public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, InventoryComponent inventory) {
         inventoryManager.giveItem(player, null, entityManager.create("Lost:antrumSabre"));
+        ProgressTrackingComponent progressTrackingComponent = new ProgressTrackingComponent();
```

Comment:
So this idea is most likely out of scope, but it could be neat to have an actual class (or later even prefabs) that define the levels to track, rather than having it hard coded in a spawn handler like this. No obligation to update, but wanted to raise the concept :-) 

(for all I know this might have already come up, in which case don't mind me repeating myself, sorry!)

Reply:
Agreed but I'd argue that this should be done in a follow-up PR